index_module.inc.php

```diff
@@ -77,38 +77,62 @@
             }
 
             // Load Mod-Config
-            $actionParameter = $request->query->get('action');
+            $actionParameter = $request->query->get('action') ?? '';
             // 1) Search $_GET['action'] in DB (field "action")
-            $menu = $db->qry_first("SELECT file, requirement FROM %prefix%menu WHERE (module = %string%) and (action = %string%)", $modParameter, $actionParameter);
+            $sqlQuery = 'SELECT `file`, `requirement` FROM `%prefix%menu` WHERE `module` = ? AND `action` = ?';
+            $menu = $database->queryWithOnlyFirstRow($sqlQuery, [$modParameter, $actionParameter]);
             if ($menu && $menu['file'] != '') {
+                // Case like
+                //  - /?mod=info2&action=change
                 if ($authentication->authorized($menu['requirement'])) {
-                    // TODO Fix security issue
```

Comment:
I suspect that this was not actually a security issue as `$menu['file']` can only be modified by somebody with administrative privileges to modify the menu entries.
And that group can include any file they want anyways, so this would not be a privilege escalation.

I suspect that the gallery and forum are places where this could become more...interesting, as we have file uploads (and access) by otherwise unprivileged users

Reply:
`$modParameter` is still a user input (and partially sanitized due to the database query).
However, having the same logic in every include scenario doesn't hurt and makes it safe anyway.

Do you see any disadvantage to do this?